src/Illuminate/Support/Str.php

```diff
@@ -875,6 +875,24 @@ public static function substrCount($haystack, $needle, $offset = 0, $length = nu
         }
     }
 
+    /**
+     * Replace text within a portion of a string.
+     *
+     * @param  string|array  $string
+     * @param  string|array  $replace
+     * @param  array|int  $offset
+     * @param  array|int|null  $length
+     * @return string|array
+     */
+    public static function substrReplace($string, $replace, $offset = 0, $length = null)
+    {
+        if ($length === null) {
+            $length = strlen($string);
+        }
```

Comment:
It looks like this conditional block isn't required since `substr_replace()` accepts a `$length` parameter value of `null` to replace the remaining substring:

![substr_replace](https://user-images.githubusercontent.com/823566/145695214-6a587df4-212a-4c00-93bb-71447a270298.png)


```php
substr_replace(
    'Laravel Framework',
    '– The PHP Framework for Web Artisans',
    8
);
// "Laravel – The PHP Framework for Web Artisans"
```

Reply:
Hey Derek, that's true, but only since PHP 8.0 is the `$length` parameter nullable. 

In PHP 7.3 and 7.4 this parameter would automatically take a value of `0` instead of `null`, causing an insert instead of a replacement of the rest of the string. So I added this conditional to keep behaviour consistent. 

The test runs pointed this out as well. 

In a newer version of Laravel (9?) we could drop the conditional. So it's perhaps an idea to add a commit to the `9.x` branch as well that removes the if-condition.